static/app/components/multiPlatformPicker.tsx

```diff
@@ -0,0 +1,300 @@
+  margin-bottom: ${space(2)};
+`;
+
+type Category = typeof PLATFORM_CATEGORIES[number]['id'];
+
+interface PlatformPickerProps {
+  addPlatform: (key: PlatformKey) => void;
+  organization: Organization;
+  platforms: PlatformKey[];
+  removePlatform: (key: PlatformKey) => void;
+  defaultCategory?: Category;
+  listClassName?: string;
+  listProps?: React.HTMLAttributes<HTMLDivElement>;
+  noAutoFilter?: boolean;
+  showOther?: boolean;
+  source?: string;
+}
+
+function PlatformPicker(props: PlatformPickerProps) {
+  const [category, setCategory] = React.useState<Category>(
+    props.defaultCategory ?? PLATFORM_CATEGORIES[0].id
+  );
+  const [filter, setFilter] = React.useState<string>(
+    props.noAutoFilter ? '' : (props.platforms[0] || '').split('-')[0]
+  );
+
+  function getPlatformList() {
+    const currentCategory = categoryList.find(({id}) => id === category);
+
+    const filterLowerCase = filter.toLowerCase();
+
+    const subsetMatch = (platform: PlatformIntegration) =>
+      platform.id.includes(filterLowerCase) ||
+      platform.name.toLowerCase().includes(filterLowerCase) ||
+      filterAliases[platform.id as PlatformKey]?.some(alias =>
+        alias.includes(filterLowerCase)
+      );
+
+    const categoryMatch = (platform: PlatformIntegration) =>
+      category === 'all' ||
+      (currentCategory?.platforms as undefined | string[])?.includes(platform.id);
+
+    const popularTopOfAllCompare = (a: PlatformIntegration, b: PlatformIntegration) => {
+      // for the all category, put popular ones at the top
+      if (category === 'all') {
+        if (isPopular(a) !== isPopular(b)) {
+          return isPopular(a) ? -1 : 1;
+        }
+      }
+      return a.id.localeCompare(b.id);
+    };
+
+    const filtered = platforms
+      .filter(filterLowerCase ? subsetMatch : categoryMatch)
+      .sort(popularTopOfAllCompare);
+
+    return props.showOther ? filtered : filtered.filter(({id}) => id !== 'other');
+  }
+
+  const platformList = getPlatformList();
+  const {addPlatform, removePlatform, listProps, listClassName} = props;
+
+  const logSearch = debounce(() => {
+    if (filter) {
+      trackAdvancedAnalyticsEvent('growth.platformpicker_search', {
+        search: filter.toLowerCase(),
+        num_results: platformList.length,
+        source: props.source,
+        organization: props.organization,
+      });
+    }
+  }, DEFAULT_DEBOUNCE_DURATION);
+
+  React.useEffect(logSearch, [filter]);
+
+  return (
+    <React.Fragment>
+      <NavContainer>
+        <CategoryNav>
+          {PLATFORM_CATEGORIES.map(({id, name}) => (
+            <ListLink
+              key={id}
+              onClick={(e: React.MouseEvent) => {
+                trackAdvancedAnalyticsEvent('growth.platformpicker_category', {
+                  category: id,
+                  source: props.source,
+                  organization: props.organization,
+                });
+                setCategory(id);
+                setFilter('');
```

Comment:
not sure if we want to clear the filter here...

Reply:
@Neo-Zhixing Just maintaining the same behavior as the existing platform picker

Reply:
I actually find the current behavior a bit confusing, but this is fine too